packages/arb-token-bridge-ui/src/pages/api/teleports/erc20.ts

```diff
@@ -0,0 +1,112 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { gql } from '@apollo/client'
+
+import {
+  getSourceFromSubgraphClient,
+  getTeleporterSubgraphClient
+} from '../../../api-utils/ServerSubgraphUtils'
+import { FetchErc20TeleportsFromSubgraphResult } from '../../../util/teleports/fetchErc20TeleportsFromSubgraph'
+
+// Extending the standard NextJs request with Deposit-params
+type NextApiRequestWithErc20TeleportParams = NextApiRequest & {
+  query: {
+    sender?: string
+    l1ChainId: string
+    page?: string
+    pageSize?: string
+  }
+}
+
+type Erc20TeleportResponse = {
+  meta?: { source: string | null }
+  data: FetchErc20TeleportsFromSubgraphResult[]
+  message?: string // in case of any error
+}
+
+export default async function handler(
+  req: NextApiRequestWithErc20TeleportParams,
+  res: NextApiResponse<Erc20TeleportResponse>
+) {
+  try {
+    const { sender, l1ChainId, page = '0', pageSize = '10' } = req.query
+
+    // validate method
+    if (req.method !== 'GET') {
+      res
+        .status(400)
+        .send({ message: `invalid_method: ${req.method}`, data: [] })
+      return
+    }
+
+    // validate the request parameters
+    const errorMessage = []
+    if (!l1ChainId) errorMessage.push('<l1ChainId> is required')
+    if (!sender) errorMessage.push('<sender> is required')
+
+    if (errorMessage.length) {
+      res.status(400).json({
+        message: `incomplete request: ${errorMessage.join(', ')}`,
+        data: []
+      })
+      return
+    }
+
+    // if invalid pageSize, send empty data instead of error
+    if (isNaN(Number(pageSize)) || Number(pageSize) === 0) {
+      res.status(200).json({
+        data: []
+      })
+      return
+    }
+
+    let subgraphClient
+    try {
+      subgraphClient = getTeleporterSubgraphClient(Number(l1ChainId))
+    } catch (error: any) {
+      // catch attempt to query unsupported networks and throw a 400
+      res.status(400).json({
+        message: error?.message ?? 'Something went wrong',
+        data: []
+      })
+      return
+    }
+
+    const subgraphResult = await subgraphClient.query({
```

Comment:
Fetching ERC20 teleports requires us to query the L1 teleported subgraph using just the sender. 
This returns ALL the erc20 teleports made by the user, irrespective of the destination chain id. 
We extract this `l3ChaiId` later in `transformTransaction` step (only for the txns we see fit to display in tx history) to avoid RPC calls.

Reply:
It's already present here: https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1677/files#diff-683d1eac78c2c5a3e4403e40eb8ced020eb0172d1f26624def5178e266e5140eR151